pkgs/by-name/ni/nixos-option/package.nix

```diff
@@ -8,9 +8,14 @@
   nix,
   nixosTests,
   shellcheck,
+  stdenv,
   stdenvNoCC,
 }:
-
+let
+  shellcheckSupported =
+    lib.meta.availableOn stdenv.buildPlatform shellcheck.compiler
+    && (builtins.tryEval shellcheck.compiler.outPath).success;
```

Comment:
tryEval is evil, can we consider not doing this if possible? e.g. ensuring that the ghc/shellcheck is marked unavailable as appropriate?

Reply:
Unfortunately this is the only way it works, see https://github.com/NixOS/nixpkgs/issues/231537#issuecomment-2525240292